typescript/src/types/misc.ts

```diff
@@ -3,3 +3,33 @@
 export type Hex = `0x${string}`;
 export type Hash = `0x${string}`;
 export type Address = `0x${string}`;
+
+export type AccessList = readonly {
+  address: Address;
+  storageKeys: readonly Hex[];
+}[];
+
+export type TransactionRequestEIP1559 = {
```

Comment:
Can we document which fields are required, optional, ignored, per the API documentation?